tests/unit/test_extractor.py

```diff
@@ -415,6 +405,33 @@ def test_upload_files_from_abfss_cognite_error(
     test_extractor.logger.error.assert_called_once()
 
 
+def test_retrieve_external_ids_from_lakehouse(test_extractor, mocker):
```

Comment:
Do we have a unit or integration test that asserts successful batching / filtering?

Reply:
Yes, `test_extractor_timeseries_service_with_state` sets the state to a timestamp in the middle of the test data set and asserts the second half of the data was written to CDF>